pkg/download/consistent_hashing.go

```diff
@@ -100,46 +102,59 @@ func (m *ConsistentHashingMode) Fetch(ctx context.Context, urlString string) (io
 		return m.FallbackStrategy.Fetch(ctx, urlString)
 	}
 
-	firstChunkResp, err := m.DoRequest(ctx, 0, m.minChunkSize()-1, urlString)
-	if err != nil {
+	br := newBufferedReader(m.minChunkSize())
+	firstReqResultCh := make(chan firstReqResult)
+	m.queue.submit(func() {
+		m.sem.Go(func() error {
```

Comment:
Same as the other comment(s). I'd like these extracted to be less code, it's a lot of code in the closure. But this can be addressed later.